plugins/storage.js

```diff
@@ -1,6 +1,32 @@
-let details = [];
-let links = [];
-let gauges = [];
+const mongoose = require("mongoose");
+
+mongoose.connect(process.env.MONGO_URI);
+
+const userDetailSchema = new mongoose.Schema({
+    userId: String,
+    name: String,
+    value: String,
+});
+userDetailSchema.index({ userId: 1, name: 1 }, { unique: true });
+const UserDetail = mongoose.model("UserDetail", userDetailSchema);
+
+const userLinkSchema = new mongoose.Schema({
+    userId: String,
+    name: String,
+    url: String,
+});
+userLinkSchema.index({ userId: 1, name: 1 }, { unique: true });
+const UserLink = mongoose.model("UserLink", userLinkSchema);
+
+const userGaugeSchema = new mongoose.Schema({
+    userId: String,
+    name: String,
+    currentName: String,
+    currentValue: Number,
+    maxValue: Number,
+});
+userGaugeSchema.index({ userId: 1, name: 1 }, { unique: true });
+const UserGauge = mongoose.model("UserGauge", userGaugeSchema);
```

Comment:
can the connection process and the schema and model definitions be moved to inside the exported function body?  doing it like this causes this to happen as a side effect of `require` which is an anti pattern, and also makes this difficult to test.